src/persist/src/file.rs

```diff
@@ -0,0 +1,197 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! In-memory implementations for testing and benchmarking.
+
+use std::collections::{HashMap, HashSet};
+use std::convert::TryInto;
+use std::fs::{File, OpenOptions};
+use std::io::{Read, Seek, SeekFrom, Write as StdWrite};
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, Mutex};
+
+use crate::error::Error;
+use crate::mem::MemSnapshot;
+use crate::persister::{Meta, Persister, Write};
+use crate::{Id, Token};
+
+/// A naive implementation of [Write] and [Meta] backed by a file.
+#[derive(Clone, Debug)]
+pub struct FileStream {
+    dataz: Arc<Mutex<File>>,
+    buf: Vec<u8>,
+}
+
+impl FileStream {
+    /// Create a new FileStream
```

Comment:
... with data stored in a file at the given path. If the file is from a previous instance of FileStream, new data is appended. (or something)